### ISSUE TITLE
Deserialization error for filter elision and tokenizer path_hierarchy

### ISSUE THREAD

**Issue by gise88** [elastic/elasticsearch-net#7870]
**Elastic.Clients.Elasticsearch version**: 8.9.1

**Elasticsearch version**: 8.8.2

**.NET runtime version**: 7.0.1

**Operating system version**:  Windows 10

**Description of the problem including expected versus actual behavior**:
Deserialization error when try to convert the stringified boolean or stringified int to the target properties 
Errors are for:
- `articles_case` field inside _filter elision_ 
- `skip` field inside _tokenizer path_hierarchy_ 
- `reverse` field inside _tokenizer path_hierarchy_ 

**Steps to reproduce**:
```
PUT _index_template/tests-bug
{
  "index_patterns": [
    "test-bug"
  ],
  "template": {
    "settings": {
      "index" : {
        "analysis" : {
          "filter" : {
            "my_elision" : {
              "type" : "elision",
              "articles" : [
                "c",
                "l"
              ],
              "articles_case" : true
            }
          },
          "tokenizer" : {
            "my_path_hierarchy" : {
              "skip" : 0,
              "reverse" : true,
              "type" : "path_hierarchy",
              "replacement" : " ",
              "delimiter" : " "
            }
          }
        }
      }
    },
    "aliases": {}
  }
}
```

whenever you `GET _index_template/tests-bug` elastisearch response is 
![image](https://github.com/elastic/elasticsearch-net/assets/6855163/86e146d8-3526-408e-8db2-b4736549a98c)

**Expected behavior**
The GetIndexTemplateAsync should deserialize and return the index template. Code:

```
var res = await client
	.Indices
	.GetIndexTemplateAsync(new GetIndexTemplateRequest(new Name("tests-bug")), CancellationToken.None);
```

It should be similar to what have been done here: https://github.com/elastic/elasticsearch-net/issues/7755

**Comment by flobernd** [comment]
Thanks for reporting! This seems to be a specification issue and indeed similar to #7755.

**Comment by gise88** [comment]
Hey @flobernd, I'm still having problem with `path_hierarchy` tokenizer.. I tried removing the `buffer_size` completely (since 1024 is the default value) or set it to string "1024" (since you added the [JsonConverter(typeof(StringifiedIntegerConverter))] attribute) but during the deserialization I get 
`The converter specified on 'Elastic.Clients.Elasticsearch.Analysis.PathHierarchyTokenizer.BufferSize' is not compatible with the type 'System.Int32'.`

![image](https://github.com/elastic/elasticsearch-net/assets/6855163/c771a5ec-c531-4c60-b21d-0363b256cbb8)

**Comment by gise88** [comment]
Hey @flobernd, do you think we might be lucky enough to have this one fixed for the next release? 🙂

**Comment by flobernd** [comment]
Hi @gise88 I'm currently on sick leave and definitely won't be able to include a fix for this in the 8.10.0 release (which I expect to happen tomorrow).

I will have another look as soon as possible. To completely fix all of these issues where a value can either be a string or another integral type, we have to wait for https://github.com/elastic/elasticsearch-specification/issues/2238 to get implemented and merged. Until this is done, we can just react on bug reports which is not very optimal.
